### PR TITLE
Twist stamped migration

### PR DESCRIPTION
**Pull request by SteveMacenski**
---

## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   |[ (add tickets here #1)](https://github.com/ros-planning/navigation2/issues/1594) |
| Primary OS tested on | Ubuntu, |
| Robotic platform tested on |  CI |
| Does this PR contain AI generated software? | None |

---

### CODE REVIEW

**Comment by Ryanf55** [comment]
The additional changes look great. Any reason I shouldn't just cherry-pick them onto the original PR?

**Comment by SteveMacenski** [comment]
> Any reason I shouldn't just cherry-pick them onto the original PR?

I figured this was easier, but if you want to continue to take the lead, I'm perfectly happy with that. I'd want you to test before merging as the primary consumer of enabling the parameters

File: nav2_behaviors/plugins/assisted_teleop.cpp [hunk 1]
```diff
@@ -128,9 +131,9 @@ ResultStatus AssistedTeleop::onCycleUpdate()
           *clock_,
           1000,
           behavior_name_.c_str() << " collided on first time step, setting velocity to zero");
-        scaled_twist.linear.x = 0.0f;
-        scaled_twist.linear.y = 0.0f;
-        scaled_twist.angular.z = 0.0f;
+        scaled_twist->twist.linear.x = 0.0f;
```
Comment (jwallace42) [inline nav2_behaviors/plugins/assisted_teleop.cpp:134]: missing header

File: nav2_behaviors/plugins/spin.cpp [hunk 2]
```diff
@@ -138,15 +138,15 @@ ResultStatus Spin::onCycleUpdate()
   double vel = sqrt(2 * rotational_acc_lim_ * remaining_yaw);
   vel = std::min(std::max(vel, min_rotational_vel_), max_rotational_vel_);
 
-  auto cmd_vel = std::make_unique<geometry_msgs::msg::Twist>();
-  cmd_vel->angular.z = copysign(vel, cmd_yaw_);
+  auto cmd_vel = std::make_unique<geometry_msgs::msg::TwistStamped>();
+  cmd_vel->twist.angular.z = copysign(vel, cmd_yaw_);
```
Comment (jwallace42) [inline nav2_behaviors/plugins/spin.cpp:142]: missing header

File: nav2_collision_monitor/src/collision_monitor_node.cpp [hunk 3]
```diff
@@ -204,11 +214,11 @@ void CollisionMonitor::publishVelocity(const Action & robot_action)
     }
   }
 
-  std::unique_ptr<geometry_msgs::msg::Twist> cmd_vel_out_msg =
-    std::make_unique<geometry_msgs::msg::Twist>();
-  cmd_vel_out_msg->linear.x = robot_action.req_vel.x;
-  cmd_vel_out_msg->linear.y = robot_action.req_vel.y;
-  cmd_vel_out_msg->angular.z = robot_action.req_vel.tw;
+  auto cmd_vel_out_msg = std::make_unique<geometry_msgs::msg::TwistStamped>();
```
Comment (jwallace42) [inline nav2_collision_monitor/src/collision_monitor_node.cpp:217]: Missing header.

**Comment by SteveMacenski** [comment]
Nice catch, looking back over it, the original had a number of issues where the proper stamps weren't stored and reused, but incorrectly restamped

**Review by jwallace42 — APPROVED** [review]
LGTM!

**Comment by Ryanf55** [comment]
I've cherry picked these changes back to the original PR. I'll report back with ArduPilot test results soon.

**Comment by SteveMacenski** [comment]
No rush - take it easy Ryan

**Pull request closed without merging**